moped-editor/src/components/GridTable/Filters.js

```diff
@@ -230,215 +129,109 @@ const Filters = ({
+    const fieldDetails = filtersConfig.fields.find(
+      (filter) => filter.name === field
+    );
+
+    // Update the field and operator values or handle when the clear X icon is clicked
+    if (fieldDetails) {
+      filtersNewState[filterIndex].field = fieldDetails.name;
+      filtersNewState[filterIndex].operator = getDefaultOperator(fieldDetails);
     } else {
-      console.debug(
-        `The filter id ${filterId} does not exist, ignoring click event.`
-      );
+      filtersNewState[filterIndex].field = null;
+      filtersNewState[filterIndex].operator = null;
     }
+
+    // Update the state
+    setFilterParameters(filtersNewState);
   };
 
   /**
    * Handles the click event on the operator drop-down
-   * @param {string} filterId - State FieldID to modify
+   * @param {string} filterIndex - filterParameters index to modify
    * @param {Object} operator - The operator object being clicked
+   * @param {string} lookupTable - The lookup table name
    */
-  const handleFilterOperatorClick = (filterId, operator) => {
-    // If the filter exists
-    if (filterId in filterParameters) {
-      // Clone state
-      const filtersNewState = { ...filterParameters };
-
-      if (operator in filtersConfig.operators) {
-        // Update Operator Value
-        filtersNewState[filterId].operator = operator;
-        // Get the GraphQL operator details
-        filtersNewState[filterId].gqlOperator =
-          filtersConfig.operators[operator].operator;
-        // Copy the envelope if available
-        filtersNewState[filterId].envelope =
-          filtersConfig.operators[operator].envelope;
-        // Copy special null value if available
-        filtersNewState[filterId].specialNullValue =
-          filtersConfig.operators[operator].specialNullValue;
-
-        // if we are switching to an autocomplete input, clear the search value
-        if (renderAutocompleteInput(filtersNewState[filterId])) {
-          filtersNewState[filterId].value = null;
-        }
-      } else {
-        // Reset operator values
-        filtersNewState[filterId].operator = null;
-        filtersNewState[filterId].gqlOperator = null;
-        filtersNewState[filterId].envelope = null;
-      }
+  const handleFilterOperatorClick = (filterIndex, operator, lookupTable) => {
+    // Clone state
+    const filtersNewState = [...filterParameters];
 
-      setFilterParameters(filtersNewState);
+    if (operator in filtersConfig.operators) {
+      // Update operator Value
+      filtersNewState[filterIndex].operator = operator;
+
+      // if we are switching to an autocomplete input, clear the search value
+      if (shouldRenderAutocompleteInput(lookupTable, operator, loading)) {
+        filtersNewState[filterIndex].value = null;
+      }
     } else {
-      console.debug(
-        `The filter id ${filterId} does not exist, ignoring click event.`
-      );
+      // Reset operator value
+      filtersNewState[filterIndex].operator = null;
```

Comment:
Maybe if the user is using the dev tools to modify the request on the way the server or Postman to create malformed ones for devious intent? I don't think it's something we need to worry about too much, but it did come to mind wrt your question.

Reply:
I'm not sure how that would work, I dont see how the operator would not be in the list of operators, since the select dropdown is populated based on that list of operators. Directly editing the operator in the url to something else doesn't affect this event handler either

Reply:
This is a really good catch because this code used to pull double duty and call `handleFilterOperatorChange` from within `handleFilterFieldMenuChange` to reset the operator input's value. So, this block was meant to handle that case. 

TLDR It used to be able to happen, but it cannot anymore. I'll remove it. 🗑️  Thanks, y'all!

Reply:
This also helped me unearth an issue when switching from existing filter with a search value like **Names contains congress** to one without a search value like **Lead is not blank**. The search value was hanging around in the search params so I made another code update. 🙏 🙏 🙏